src/WebJobs.Script.WebHost/Middleware/JobHostEasyAuthMiddleware.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Azure.AppService.Middleware.AspNetCoreMiddleware;
+using Microsoft.Azure.WebJobs.Script.Middleware;
+using Microsoft.Azure.WebJobs.Script.WebHost.Configuration;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Middleware
+{
+    public class JobHostEasyAuthMiddleware : IJobHostHttpMiddleware
+    {
+        private readonly ILogger<JobHostEasyAuthMiddleware> _logger;
+        private readonly IConfiguration _configuration;
+        private RequestDelegate _invoke;
+
+        public JobHostEasyAuthMiddleware(IOptions<HostEasyAuthOptions> hostEasyAuthOptions, ILogger<JobHostEasyAuthMiddleware> logger, IConfiguration configuration)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _configuration = configuration ?? throw new ArgumentNullException(nameof(logger));
+
+            RequestDelegate contextNext = async context =>
+            {
+                if (context.Items.Remove(ScriptConstants.EasyAuthMiddlewareRequestDelegate, out object requestDelegate) && requestDelegate is RequestDelegate next)
+                {
+                    await next(context);
+                }
+            };
+            if (hostEasyAuthOptions.Value.SiteAuthEnabled)
+            {
+                var easyAuthMiddleware = new EasyAuthMiddleware(contextNext, _configuration);
+                _invoke = easyAuthMiddleware.InvokeAsync;
+                _logger.LogDebug($"{nameof(JobHostEasyAuthMiddleware)} Creating {nameof(EasyAuthMiddleware)}");
```

Comment:
Is this going to be too noisy?

Reply:
I think this had originally been included for debugging purposes - it would make sense to clean it up.